### PR TITLE
8340213: jcmd VM.events ignores max argument

### PR DESCRIPTION
**Pull request by lmesnik**
The inner  'int max;' declaration hide previous max.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Warning
&nbsp;⚠️ Found leading lowercase letter in issue title for `8340213: jcmd VM.events ignores max argument`

### Issue
 * [JDK-8340213](https://bugs.openjdk.org/browse/JDK-8340213): jcmd VM.events ignores max argument (**Bug** - P4)

### Reviewers
 * [Sonia Zaldana Calles](https://openjdk.org/census#szaldana) (@SoniaZaldana - Committer)
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**)
 * [Alex Menkov](https://openjdk.org/census#amenkov) (@alexmenkov - **Reviewer**)
 * [Hamlin Li](https://openjdk.org/census#mli) (@Hamlin-Li - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21024/head:pull/21024` \
`$ git checkout pull/21024`

Update a local copy of the PR: \
`$ git checkout pull/21024` \
`$ git pull https://git.openjdk.org/jdk.git pull/21024/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21024`

View PR using the GUI difftool: \
`$ git pr show -t 21024`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21024.diff">https://git.openjdk.org/jdk/pull/21024.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21024#issuecomment-2353730736)

### CODE REVIEW

File: test/hotspot/jtreg/serviceability/dcmd/vm/EventsTest.java [hunk 1]
```diff
@@ -64,9 +68,26 @@ public void run_selected(CommandExecutor executor) {
...
+        OutputAnalyzer output = executor.execute("VM.events max=" + MAX);
+        long lines = output.asLines().stream().filter(x -> x.contains("Loading class")).count();
+        Assert.assertTrue(lines == MAX, "There are should be " + MAX + " lines");
+    }
+
+    public void run_max_selected(CommandExecutor executor) {
+        OutputAnalyzer output = executor.execute("VM.events log=load max=" + MAX);
+        long lines = output.asLines().stream().filter(x -> x.contains("Loading class")).count();
+        Assert.assertTrue(lines == MAX, "There are should be " + MAX + " lines");
+        output.stdoutShouldNotMatch(buildHeaderPattern("Events"));
```
Comment (SoniaZaldana) [inline test/hotspot/jtreg/serviceability/dcmd/vm/EventsTest.java:81]: Could also check the output contains the selected event category like in `run_selected` above.

**Review by SoniaZaldana — APPROVED** [review]
I am not a Reviewer but this looks good to me.

File: test/hotspot/jtreg/serviceability/dcmd/vm/EventsTest.java [hunk 2]
```diff
@@ -38,6 +39,9 @@
  */
 public class EventsTest {
 
+    // MAX should be less then number of actually recored events
```
Comment (SoniaZaldana) [inline test/hotspot/jtreg/serviceability/dcmd/vm/EventsTest.java:42]: Nit, spelling for `than` and `recorded`.

File: test/hotspot/jtreg/serviceability/dcmd/vm/EventsTest.java (hunk 1 above)
Comment (lmesnik) [reply test/hotspot/jtreg/serviceability/dcmd/vm/EventsTest.java:82]: Thanks! done.

**Review by SoniaZaldana — APPROVED** [review]
Thanks!

**Review by plummercj — APPROVED** [review]
Looks good. Thanks for adding the new test cases.

**Review by alexmenkov — APPROVED** [review]

**Review by Hamlin-Li — APPROVED** [review]

**Comment by kevinjwalls** [comment]
If this is not too late:
it's odd that max is a STRING parameter, and the command does the conversion itself.  Is it possible to change to an INT parameter?  I can log it for the future if this change is all tested and ready to go.

**Comment by lmesnik** [comment]
I filed separate  https://bugs.openjdk.org/browse/JDK-8340334  for changing type. So any users of jcmd can track it, and to separate the backports.

**Pull request closed without merging**